pallets/nodle-staking/src/lib.rs

```diff
@@ -0,0 +1,2311 @@
+
+            let nominator_free_balance = T::Currency::free_balance(&nominator);
+            ensure!(
+                nominator_free_balance >= nominations.total,
+                Error::<T>::InsufficientBalance
+            );
+
+            // T::Currency::reserve(&nominator, more)?;
+            T::Currency::set_lock(
+                STAKING_ID,
+                &nominator,
+                nominations.total,
+                WithdrawReasons::all(),
+            );
+
+            let before = validator_state.total;
+            validator_state.inc_nominator(nominator.clone(), more);
+            let after = validator_state.total;
+            <Total<T>>::mutate(|x| *x += more);
+            if validator_state.is_active() {
+                Self::update_validators_pool(validator.clone(), validator_state.total);
+            }
+            <ValidatorState<T>>::insert(&validator, validator_state);
+            <NominatorState<T>>::insert(&nominator, nominations);
+            Self::deposit_event(Event::NominationIncreased(
+                nominator, validator, before, after,
+            ));
+            Ok(().into())
+        }
+        /// Bond less for nominators with respect to a specific nominated validator
+        #[pallet::weight(T::WeightInfo::nominator_bond_less())]
+        pub fn nominator_bond_less(
+            origin: OriginFor<T>,
+            validator: T::AccountId,
+            less: BalanceOf<T>,
+        ) -> DispatchResultWithPostInfo {
+            let nominator = ensure_signed(origin)?;
+            let mut nominations =
+                <NominatorState<T>>::get(&nominator).ok_or(Error::<T>::NominatorDNE)?;
+            let remaining = nominations
+                .dec_nomination(validator.clone(), less)
+                .ok_or(Error::<T>::NominationDNE)?
+                .ok_or(Error::<T>::Underflow)?;
+            ensure!(
+                remaining >= T::MinNomination::get(),
+                Error::<T>::NominationBelowMin
+            );
+            ensure!(
+                nominations.active_bond >= T::MinNominatorStake::get(),
+                Error::<T>::NominatorBondBelowMin
+            );
+
+            ensure!(
+                nominations.unlocking.len() < MAX_UNLOCKING_CHUNKS,
+                Error::<T>::NoMoreChunks,
+            );
+
+            let mut validator_state =
+                <ValidatorState<T>>::get(&validator).ok_or(Error::<T>::ValidatorDNE)?;
+
+            // T::Currency::unreserve(&nominator, less);
+            nominations.unlocking.push(UnlockChunk {
+                value: less,
+                session_idx: Self::active_session() + T::BondedDuration::get(),
+            });
+
+            let before = validator_state.bond + validator_state.nomi_bond_total;
+            validator_state.dec_nominator(nominator.clone(), less);
+            let after = validator_state.bond + validator_state.nomi_bond_total;
+            <Total<T>>::mutate(|x| *x -= less);
```

Comment:
use safe math or saturating operators :)

Reply:
:+1:  Moved to saturating_sub()